app/views/exception_notifier/_session.text.erb

```diff
@@ -0,0 +1,20 @@
+<% browser = Browser.new(@request.user_agent) %>
+Browser Name: <%= browser.name %>
+Browser Version: <%= browser.full_version %>
+Browser Platform: <%= browser.platform %> <%= browser.platform.version %>
+Browser Device: <%= browser.device.name %>
+Browser Bot?: <%= browser.bot? %>
+
+Referer: <%= @request.referer %>
+
+<% session = @request.session.to_hash %>
```

Comment:
Is there any possibility of us transmitting any PII in any dangerous way here?

Reply:
This line shouldn't render anything (Since the prefix is `<%` instead of `<%=`).

Where we need to look out is lower on line 16 where we're filtering out the things that will contain PII.

Reply:
Thanks Jonathan. I get that we're just casting it here, but this is where I had the thought pop up that we might be exposing more than we want to.

Reply:
Thanks for being vigilant. You reminded me that we do store the user's unconfirmed phone in the session when they first set up 2FA, so we should filter that out. I will also check the ID verification flow to make sure we don't capture any PII there.